presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -231,7 +245,9 @@ public PrestoSparkTaskExecutorFactory(
             PrestoSparkConfig prestoSparkConfig,
             BlockEncodingSerde blockEncodingSerde,
             NativeExecutionProcessFactory processFactory,
-            NativeExecutionTaskFactory taskFactory)
+            NativeExecutionTaskFactory taskFactory,
+            PrestoSparkShuffleInfoFactory<?, ?> shuffleInfoFactory,
+            PrestoSparkShuffleInfoSerializer shuffleInfoSerializer)
```

Comment:
I am wondering if we could combine serializer with above factory and change the name of PrestoSparkShuffleInfoFactory to PrestoSparkShuffleInfoTranslator that essentially is responsible for translating shuffle info to different representations. 
Currently we have too many such classes that are very confusing. This will make our life easier later for maintenance.